frontend/src/views/pages/inventory/InventoryAdd.vue

```diff
@@ -0,0 +1,226 @@
+                >
+                  <template #prepend-content>
+                    <CIcon name="cil-calendar"/>
+                  </template>
+                  <template #label>
+                    {{ property.label }}
+                  </template>
+                  <template #input>
+                    <masked-input
+                      type="text"
+                      :name="property.key"
+                      class="form-control"
+                      :mask="[/0|1|2|3/, /\d/, '-', /0|1|2|3/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]"
+                      :guide="true"
+                      placeholderChar="_"
+                      :showMask="true"
+                      :keepCharPositions="true"
+                      :required="property.isRequired"
+                    />
+                  </template>
+                </CFormGroup> -->
+              </div>
+              <CButton color="primary" type="submit">Add vehicle to the inventory</CButton>
+            </CForm>
+          </CCardBody>
+        </CCard>
+      </CCol>
+      <CCol></CCol>
+    </CRow>
+    <CModal
+      :show.sync="showingModal"
+      :centered="true"
+    >
+      <p>Do you wish to add another vehicle?</p>
+      <template #header>
+        <h6 class="modal-title">Vehicle added successfully!</h6>
+        <CButtonClose @click="showingModal = false" />
+      </template>
+      <template #footer>
+        <CButton @click="showingModal = false" color="success">Yes</CButton>
+        <router-link to="/inventory"><CButton color="danger">No</CButton></router-link>
+      </template>
+    </CModal>
+  </div>
+</template>
+
+<script>
+const axios = require('axios');
+const { containsPermissions } = require("../../../utils/index");
+import MaskedInput from 'vue-text-mask'
+
+export default {
+  name: 'InventoryAdd',
+  data() {
+    return {
+      dealership: null,
+      vehicleProperties: null,
+      errorMessage: null,
+      showingModal: false
+    }
+  },
+  methods: {
+    submitForm(form) {
+      let properties = {};
+      this.vehicleProperties.forEach(property => {
+        if (!form.target.elements[property.key]) {
+          properties[property.key] = null;
+        } else {
+          if (property.inputType == "List") {
+            const itemString = form.target.elements[property.key].value;
+            const items = itemString.split(";");
+            properties[property.key] = items;
+          } else if (property.inputType == "Date") {
+            properties[property.key] = form.target.elements[property.key].value
+          } else if (property.inputType == "Number") {
+            properties[property.key] = form.target.elements[property.key].value
+          } else if (property.inputType == "Currency") {
+            properties[property.key] = form.target.elements[property.key].value
+          } else {
+            properties[property.key] = form.target.elements[property.key].value
+          }
+        }
+      })
```

Comment:
The field formatting seems to be used in multiple files. This is could be refactored to be extracted and referenced.

Reply:
This has been fixed

Reply:
Yeah I confirm. it looks better now